bokeh/document.py

```diff
@@ -298,6 +300,12 @@ def __init__(self, **kwargs):
         self._callbacks = {}
         self._session_callbacks = {}
         self._session_context = None
+        self._modules = []
+
+    def __del__(self):
+        for module in self._modules:
+            if module.__name__ in sys.modules:
+                del sys.modules[module.__name__]
```

Comment:
`__del__` is extremely uncommon I think a note here about why this is where it is would help anyone else looking at this code